### PR TITLE
Fix: Allow more than one type of node structure

### PR DESCRIPTION
**Pull request by mdotwills**
Changes:

- Relaxed `path` property of node type

### CODE REVIEW

File: src/helpers/propTypes/TreePickerPropTypes.js [hunk 1]
```diff
@@ -2,19 +2,27 @@ import { PropTypes } from 'react';
...
+      id: PropTypes.string.isRequired,
+      isExpandable: PropTypes.bool,
+      label: PropTypes.string.isRequired,
+      type: PropTypes.string.isRequired,
+    }),
+    node: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      isExpandable: PropTypes.bool,
+      label: PropTypes.string.isRequired,
+      path: PropTypes.arrayOf(
```
Comment (OndrejCholeva) [inline src/helpers/propTypes/TreePickerPropTypes.js]: What is the reason of having two almost identical node types? What about just making path optional?

Comment (mdotwills) [inline src/helpers/propTypes/TreePickerPropTypes.js]: ^ much cleaner solution, thought there was a need for explicit `path` on some components but we can do without it.

**Comment by omgaz** [comment]
I don't see why a node would not have a path? Can you explain?

**Comment by mdotwills** [comment]
After offline discussion, we've decided to keep the `path` property, regardless if used explicitly, as a required property as it is fundamental to the tree picker structure concept.

**Pull request closed without merging**